### ISSUE TITLE
How to create a workunit using the command python pythonfile.py 

### ISSUE THREAD

**Issue by xFrancisc0** [marius311/boinc2docker#11]
Good afternoon,

My main problem is that i'm trying to create a WU using a source file called fibonacci.py located in the BOINC server, typing this command:

`bin/boinc2docker_create_work.py python:alpine python fibonacci.py`

However, when i connect the BOINC client to the server, it downloads and executes and then upload the WU without problems, however when i go to see the results, it's not valid.

The results logs for the command mentioned above in hostname/projectname_ops/db_action.php?table=result&id=1 is

`2020-07-10 02:41:25 (11808): Guest Log: Running...`

`2020-07-10 02:41:25 (11808): Guest Log: python: can't open file '/home/boincadm/project/pruebas/fibonacci.py': [Errno 2] No such file or directory`

`2020-07-10 02:41:25 (11808): Guest Log: boinc_app exited (2)`

Its not finding the file fibonacci.py to compile and execute on the BOINC client.

I'd like to create workunits using python compile/execute command from source files, like:

`boinc2docker_create_work.py python:alpine python pythonfile.py`

I saw an issue from another user that he tried to add files and input files on boinc2docker_create_work.py but I didn't understand it, because I've been using:

`boinc2docker_create_work.py python:alpine python (commands)`

instead of:

`boinc2docker_create_work.py(parameter1, parameter2, ..., parameter n)`

**Comment by marius311** [comment]
Hi, as currently set up, no extra files are included in the workunit except for the Docker image itself, so your file `fibonacci.py` does not end up on the user's computer. The way to do this is create a new image which includes this file it (which would be super trivial, basically just a Dockerfile with `FROM python:alpine` and then a `COPY fibonacci.py ...`) and submit the job pointing to that new image.

**Comment by xFrancisc0** [comment]
Hello!

I've been searching for information and I found how to use:

`boinc2docker_create_work.py(parameter1, parameter2, ..., parameter n)`

at this topic: https://github.com/marius311/boinc-server-docker/issues/18



I found on this topic: https://github.com/marius311/boinc2docker/issues/10 that user **gateware2** wanted to execute:

`boinc2docker_create_work("python:3-slim", "prim.py", [("input.gz", "input.txt", ['gzip','nodelete'])])`

command and the file prim.py was a python source file, just as fibonacci.py, without using DockerFile.

Questions:
1.- Couldn't I directly execute fibonacci.py just as **gateware2** tried to execute prim.py? Or is it necessary to create the new Image which includes fibonacci.py for its execution on BOINC clients?

2.- How can i submit a job pointing to a custom Docker Image using boinc2docker? I'm working with boinc-server-docker.


Thanks for your help, time and patience

**Comment by marius311** [comment]
>1.- Couldn't I directly execute fibonacci.py just as gateware2 tried to? Or is it necessary to create the new Image which includes fibonacci.py for its execution on BOINC clients?

You could, but as long as this code would be used in all of your jobs, I think including your code into the Docker image is the easier approach to be honest.

> 2.- How can i submit a job pointing to a custom Docker Image using boinc2docker?

Just get the image on to the computer running the BOINC server somehow (either build it on the server, or build it elsewhere and `docker pull` it from the machine running your server) then its just `boinc2docker_create_work.py your_custom_image_name`

**Comment by xFrancisc0** [comment]
Thanks! I'll try doing that.

If everything works fine I'll post my results in here to give my personal feedback.

Best regards from Chile, South America.

**Comment by xFrancisc0** [comment]
It worked! Thanks for everything @marius311 

Feedback results
===========

1.- I executed
`docker pull python:alpine`

2.- I created a folder called workunit_source where i put the Dockerfile and fibonacci.py files. This folder i put it on the boinc-server-docker's apache container. 

Dockerfile's data was:

`FROM python:alpine`
`RUN mkdir /workunit_source`
`COPY --chown=1000 fibonacci.py /workunit_source/fibonacci.py`



3.- I went to the folder workunit_source and executed

`docker build -t fibonacci_image . ` 
(The dot means current folder)


4.- I executed

`docker run fibonacci_image python /workunit_source/fibonacci.py` and it worked


5.- I executed

`bin/boinc2docker_create_work.py fibonacci_image python /workunit_source/fibonacci.py` and it worked too. I went to the results section on project_ops and it worked and returned the value i wanted.
